src/control/provider/system/system.go

```diff
@@ -0,0 +1,30 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package system
+
+type Provider interface {
```

Comment:
Possibly -- though in hindsight I'm not sure that this interface should even be defined here. The general rule of thumb is to define interfaces where they're used rather than next to their implementations.

Reply:
I think it's okay to be here for the time being. We can always move it later if only one package uses it.

Reply:
Thinking about this a bit more: This package isn't meant to be storage-specific. It's meant to provide a shim for system functionality that we have to mock in tests. That having been said, a single system.Provider interface (which would presumably continue to grow) gets us heading back in the direction of server.External, which is what I wanted to avoid. I've replaced this single interface with a series of single-function interfaces which can be used in a compose-able way similarly to how the stdlib does things.